src/main/java/net/minecraftforge/client/model/CompositeModel.java

```diff
@@ -125,6 +125,12 @@ public Collection<Material> getMaterials(IGeometryBakingContext context, Functio
         return textures;
     }
 
+    @Override
+    public Set<String> getComponentNames(boolean recursive)
+    {
+        return children.keySet();
```

Comment:
Is it not possible that a composite model is make up out of another composite model?